source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift

```diff
@@ -0,0 +1,96 @@
+                valueView.labelText.textColor = textColor
+            }
+            
+            if !(titleView.titleText.labelText.stringValue.isEmpty) || !(valueView.labelText.stringValue.isEmpty) {
+                titleStack.addArrangedSubview(titleView.titleText)
+                valueStack.addArrangedSubview(valueView)
+            }
+        }
+        
+        // Add both elements into the Horizontal Stack
+        horizontalStack.addArrangedSubview(titleStack)
+        horizontalStack.addArrangedSubview(valueStack)
+        
+        // Add constraints once all the views are in the same view heirarchy
+        let horizontalStackViews = horizontalStack.arrangedSubviews
+        guard let titleStackagain = horizontalStackViews[0] as? NSStackView else { return NSView() }
+        guard let valueStackagain = horizontalStackViews[1] as? NSStackView else { return NSView() }
+        
+        titleStackagain.leadingAnchor.constraint(equalTo: horizontalStack.leadingAnchor).isActive = true
+        titleStackagain.topAnchor.constraint(equalTo: horizontalStack.topAnchor).isActive = true
+        titleStackagain.bottomAnchor.constraint(equalTo: horizontalStack.bottomAnchor).isActive = true
+        // Spacing between title and value in the horizontal Stack
+        titleStackagain.trailingAnchor.constraint(equalTo: valueStackagain.leadingAnchor, constant: -10).isActive = true
+        // Getting Max width from Host config if it exists
+        titleStackagain.widthAnchor.constraint(lessThanOrEqualToConstant: CGFloat(truncating: factsetConfig?.title.maxWidth ?? 150)).isActive = true
+
+        valueStackagain.trailingAnchor.constraint(equalTo: horizontalStack.trailingAnchor).isActive = true
+        valueStackagain.topAnchor.constraint(equalTo: horizontalStack.topAnchor).isActive = true
+        valueStackagain.bottomAnchor.constraint(equalTo: horizontalStack.bottomAnchor).isActive = true
+
+        // Make the height of each title and value equal
+        for (index, elem) in titleStackagain.arrangedSubviews.enumerated() {
+            guard let titleView = elem as? ACRFactTextField else { return ACRFactTextField() }
+            let valueArray = valueStackagain.arrangedSubviews
+            guard let valueView = valueArray[index] as? ACRFactTextField else { return ACRFactTextField() }
+            titleView.heightAnchor.constraint(equalTo: valueView.heightAnchor).isActive = true
+        }
+        
+        return horizontalStack
+    }
+}
+
+// MARK: Class for Title Properties
+class TitleFact: ACRFactTextField {
+    var titleText = ACRFactTextField()
+    
+    override init() {
+        super.init()
+        setupTitle()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setupTitle() {
+        // TODO: Make it get all properties from host config
+        titleText.labelText.font = NSFont.boldSystemFont(ofSize: 12)
```

Comment:
Is this whole class needed?
Guess `ACRFactTextField` can be reused with another method `setTitle(_ title: String)`

Reply:
Yeah true
Will do that I guess